src/main/java/com/minecolonies/coremod/entity/ai/citizen/CitizenAI.java

```diff
@@ -133,7 +133,7 @@ private IState calculateNextState()
             }
 
             // Sick
-            if (citizen.getCitizenDiseaseHandler().isSick() && citizen.getCitizenJobHandler().getColonyJob().canAIBeInterrupted())
+            if (citizen.getCitizenDiseaseHandler().isSick() && citizen.getLastHurtMob() == null)
```

Comment:
dont think this makes a lot of sense, AI interrupted is made for this purpose, now crafters are gonna stop mid craft, guards stop immediately after fighting even when there is more mobs around

Reply:
This is only in the guard branch, so it doesnt affect crafters here.

Reply:
The thing is, I believe AIInterrupt for eating is fine on the strictness, but I think, especially for guards, the strictness is too high (Especially the super branch of the guard).

Reply:
what is wrong with AI interrupt here to begin with?

Reply:
They run around for a while still while sick.

Reply:
well they have a combat timer for not immediately stopping fighting exactly for that reason they're supposed to keep fighting and not run away in the middle and die